File: 1572.matrix-diagonal-sum/solution.py
class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        N = len(mat)
        s1 = 0
        s2 = 0
        for i in range(N):
            s1 += mat[i][i]
            s2 += mat[i][N - i - 1] if i != N - i - 1 else 0
        return s1 + s2

File: 1572.matrix-diagonal-sum/solution.kt
class Solution {
    fun diagonalSum(mat: Array<IntArray>): Int {
        val N = mat.size
        var s1 = 0
        var s2 = 0
        for (i in 0 until N) {
            s1 += mat[i][i]
            s2 += when(N - i - 1) {
                i -> 0
                else -> mat[i][N - i - 1]
            }
        }
        return s1 + s2
    }
}
